microprofile-metrics/src/test/java/org/jboss/eap/qe/microprofile/metrics/integration/config/CustomMetricCustomConfigSourceProviderTest.java

```diff
@@ -0,0 +1,150 @@
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URL;
+
+import org.apache.commons.io.FileUtils;
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.container.test.api.RunAsClient;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.as.arquillian.api.ServerSetup;
+import org.jboss.as.arquillian.api.ServerSetupTask;
+import org.jboss.as.arquillian.container.ManagementClient;
+import org.jboss.as.controller.PathAddress;
+import org.jboss.as.controller.client.ModelControllerClient;
+import org.jboss.as.controller.client.helpers.ClientConstants;
+import org.jboss.as.controller.operations.common.Util;
+import org.jboss.dmr.ModelNode;
+import org.jboss.eap.qe.microprofile.tooling.server.configuration.creaper.ManagementClientProvider;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.EmptyAsset;
+import org.jboss.shrinkwrap.api.exporter.ZipExporter;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.runner.RunWith;
+import org.wildfly.extras.creaper.commands.modules.AddModule;
+import org.wildfly.extras.creaper.commands.modules.RemoveModule;
+
+@RunWith(Arquillian.class)
+@RunAsClient
+@ServerSetup(CustomMetricCustomConfigSourceProviderTest.SetupTask.class)
+public class CustomMetricCustomConfigSourceProviderTest extends CustomMetricBaseTest {
+    private static final PathAddress SYSTEM_PROPERTY_ADDRESS = PathAddress.pathAddress()
+            .append(SYSTEM_PROPERTY, CustomConfigSource.FILEPATH_PROPERTY);
+    private static final String PROPERTY_FILENAME = "custom-metric.properties";
+
+    private byte[] bytes;
+
+    File propertyFile = new File(CustomMetricCustomConfigSourceProviderTest.class.getResource(PROPERTY_FILENAME).getFile());
+
+    @Deployment(testable = false)
+    public static WebArchive createDeployment() {
+        WebArchive webArchive = ShrinkWrap
+                .create(WebArchive.class, CustomMetricCustomConfigSourceProviderTest.class.getSimpleName() + ".war")
+                .addClasses(CustomCounterIncrementProvider.class, CustomCounterMetric.class, CustomMetricService.class,
+                        CustomMetricApplication.class, CustomMetricAppInitializer.class)
+                .addAsWebInfResource(EmptyAsset.INSTANCE, "beans.xml");
+        return webArchive;
+    }
+
+    @Before
+    public void backup() throws IOException {
+        bytes = FileUtils.readFileToByteArray(propertyFile);
+    }
+
+    @After
+    public void restore() throws IOException {
+        FileUtils.writeByteArrayToFile(propertyFile, bytes);
+    }
+
+    void setConfigProperties(int increment) throws IOException {
+        FileUtils.writeStringToFile(propertyFile, INCREMENT_CONFIG_PROPERTY + "=" + Integer.toString(increment));
+    }
+
+    public static class SetupTask implements ServerSetupTask {
```

Comment:
Please describe the purpose of this SetupTask.

Reply:
SetupTask is specific to the scenario. `setup` method has few self-descriptive lines.